### PR TITLE
Add spec on orders

### PR DESCRIPTION
**Pull request by jhs7jhs**

### CODE REVIEW

File: spec/Order.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+CodeChain supports an on-chain order to support DEX(Decentralized Exchange).
+
+## Order
+
+An order can be either a _point-to-point order_ between a maker and a taker, or a _broadcast order_ between a maker, a relayer and takers.
+Point-to-point orders allow two parties(makers and takers) to directly exchange assets between each other.
+Broadcast orders are similar to point-to-point orders, but also allows relayers to take fee assets from makers and takers. Broadcast orders are usually used in decentralized exchange platforms.
```
Comment (ScarletBlue) [inline spec/Order.md]: Broadcast orders are similar to point-to-point orders, but also allows relayers to take fee assets from makers and takers. Broadcast orders are usually used in decentralized exchange platforms. -> Broadcast orders are similar to point-to-point orders, but also allow relayers to take fee assets from makers and takers. Broadcast orders are usually used in decentralized exchange platforms.

File: spec/Order.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+Orders on CodeChain is implemented on UTXO forms.
```
Comment (ScarletBlue) [inline spec/Order.md]: CodeChain are

File: spec/Order.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+Basically, an order is like a tag on products in grocery stores. Orders can be put on inputs and outputs of transfer transactions. If an input/output has an order, it means that specific input/output should be exchanged as its order says. Think about this situation: Let’s say we have an order to exchange 10 gold to 100 silver, and we've put that order on a 10-gold input. Then there should be a 100-silver output with the same order of 10 gold to 100 silver. If there isn’t enough gold, there would perhaps be a 5-gold output and a 50-silver output, both with the same order, or, whatever outputs with the same order while maintaining the order equilibrium.
```
Comment (ScarletBlue) [inline spec/Order.md]: it means that specific input/output -> it means that that specific input/output

File: spec/Order.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+Assets with orders must be able to be spent by takers or relayers without any permission. But in the CodeChain UTXO form, those parties should provide a lock script and an unlock script to prove the ownership of the assets. To solve the problem, CodeChain runs VM on **orders**, not *transactions*, if there are orders on inputs. If there are no orders on inputs, CodeChain runs VM on transactions as usual. Partial signing is not allowed on transactions with orders. With this convention, takers and relayers can take the ownership of the assets with orders with unlock scripts which are provided by makers.
```
Comment (ScarletBlue) [inline spec/Order.md]: To solve the problem, CodeChain runs VM on **orders**, not *transactions*, if there are orders on inputs. -> In order to solve the problem, if there are orders on inputs, CodeChain runs VM on **orders**, not *transactions*.

File: spec/Order.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+|---------------|-----------------------|
+| networkId     | NetworkId             |
+| burns         | AssetTransferInput[]  |
+| inputs        | AssetTransferInput[]  |
+| outputs       | AssetTransferOutput[] |
+| orders        | OrderOnTransfer[]     |
+
+## How to support partial fills?
+
+As described before, with the 10-gold-to-100-silver order, it is possible to make a transfer transaction which has a 10-gold input that results in a 5-gold and 50-silver output, tagged with the same order. (Other inputs/outputs should be provided by a taker or a relayer). After this transaction, an asset contains the hash of the *consumed order*, which is 5-gold-to-50-silver order, not 10-gold-to-100-silver order. In order to use the 5-gold output, provide the 5-gold-to-50-silver order with the same information except for the `assetAmountFrom` and the `assetAmountTo` field. Neither a lock script nor a unlock script is needed to the 5-gold input. CodeChain will compare the order of an input and the order of a corresponding previous output, and will not run VM on the order if those orders are same.
```
Comment (ScarletBlue) [inline spec/Order.md]: As described before, with the 10-gold-to-100-silver order, -> As described above in the 10-gold-to-100-silver order,

Comment (ScarletBlue) [reply spec/Order.md]: Neither a lock script nor a unlock script is needed to the 5-gold input. -> Neither a lock script nor an unlock script is needed for the 5-gold input.

Comment (ScarletBlue) [reply spec/Order.md]: CodeChain will compare the order of an input and the order of a corresponding previous output, and will not run VM on the order if those orders are same. -> CodeChain will compare the order of an input and the order of the corresponding previous output, and will not run VM on the order if those orders are identical.

**Comment by jhs7jhs** [comment]
@ScarletBlue Fixed!

**Pull request merged**